Rewrite comment that explains and illustrates neighbors_to and _of. Also get rid of the size diff number.

ursg opened the pull request:
The old one was slightly confusing, and anyway no longer matching the behaviour with the new neighbor definitions.

There was also the weird size ratio value returned for every neighborhood value, that we never made use of. That one's gone now.

ursg committed:
Rewrite comment that explains and illustrates neighbors_to and _of

The old one was slightly confusing, and anyway no longer matching the
behaviour with the new neighbor definitions.

markusbattarbee commented on dccrg.hpp:
```diff
@@ -7081,18 +7082,12 @@ template <
-	neighbors_to[5] is empty because neighbors[5] = 1, 6, 9, 10
+	As a reverse of neighbors_to, this data structure contains the cells that
+	consider a given cell *as their neighbors*, as well as their offsets and
+	refinement level difference.
```
The fourth value is not exactly a ref level diff. 
https://github.com/fmihpc/dccrg/blame/cad44660c86a904ecf35061c92cc744e9ad4f37e/dccrg.hpp#L4912

Looks like it can be 2, 1, or -2? Ecch.

ursg replied:
Right, I guess the idea here was to encode the cell size ratio in that number. If they are the same size, it is 1. If they are larger, it is 2.

A smaller cell would be 0.5, of course. But this is an integer variable. So it is encoded as -2.

Horrible. :)

ursg committed:
Further clarify cell size ratio number in comment

ursg committed:
Actually return size difference of neighbors_to correctly.

Not sure if anybdy ever really looks at these numbers. But while we're
fixing the comments about them, we might as well get their numbers
right.

ursg committed:
Replace confusing bitshifts by a hardcoded "2"

ursg committed:
Get rid of neighbor cell size difference return val altogether.

ursg committed:
Remove some more dangling references to neighbor denominators.

ursg commented:
Soo, this appears to work now. Let's plonk it into the parent PR.
